pool/sqrt_price_math.gno

```diff
@@ -6,46 +6,61 @@ import (
-	requireUnsigned(liquidity, ufmt.Sprintf("[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromAmount0RoundingUp() || liquidity(%d) >= 0", liquidity))
-	requireUnsigned(amount, ufmt.Sprintf("[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromAmount0RoundingUp() || amount(%d) >= 0", amount))
+	requireUnsigned(
+		sqrtPX96,
+		ufmt.Sprintf(
+			"[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromAmount0RoundingUp() || sqrtPX96(%d) >= 0",
+			sqrtPX96,
+	))
+
+	requireUnsigned(
+		liquidity,
+		ufmt.Sprintf(
+			"[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromAmount0RoundingUp() || liquidity(%d) >= 0",
+			liquidity,
+	))
+
+	requireUnsigned(
+		amount,
+		ufmt.Sprintf(
+			"[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromAmount0RoundingUp() || amount(%d) >= 0",
+			amount,
+	))
+
 	if amount == 0 {
 		return sqrtPX96
 	}
 
 	numerator1 := liquidity << 96
-	requireUnsigned(numerator1, ufmt.Sprintf("[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromAmount0RoundingUp() || numerator1(%d) >= 0", numerator1))
-
 	product := amount * sqrtPX96
-	requireUnsigned(product, ufmt.Sprintf("[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromAmount0RoundingUp() || product(%d) >= 0", product))
-
-	var denominator bigint
 
 	if add {
-		if product/amount == sqrtPX96 {
-			denominator = numerator1 + product
-			requireUnsigned(denominator, ufmt.Sprintf("[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromAmount0RoundingUp() || denominator(%d) >= 0", denominator))
-
-			if denominator >= numerator1 {
-				requireUnsigned(numerator1*sqrtPX96/denominator, ufmt.Sprintf("[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromAmount0RoundingUp() || numerator1 * sqrtPX96 / denominator(%d) >= 0", numerator1*sqrtPX96/denominator))
-				return numerator1 * sqrtPX96 / denominator
-			}
-		}
-
-		requireUnsigned(numerator1/((numerator1/sqrtPX96)+amount), ufmt.Sprintf("[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromAmount0RoundingUp() || numerator1 / ( (numerator1 / sqrtPX96) + amount(%d) >= 0", (numerator1/((numerator1/sqrtPX96)+amount))))
-		return numerator1 / ((numerator1 / sqrtPX96) + amount)
+		denominator := numerator1 + product
+
+		require(
+			denominator != 0,
+			ufmt.Sprintf(
+				"[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromAmount0RoundingUp() || expected denominator != 0, got: %d",
+				denominator,
+		))
+
+		return numerator1 * sqrtPX96 / denominator
 	}
-	require(product/amount == sqrtPX96 && numerator1 > product, "[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromAmount0RoundingUp() || condition must return true")
 
-	denominator = numerator1 - product
-	requireUnsigned(denominator, ufmt.Sprintf("[POOL] sqrt_price_math.gno__sqrtPriceMathGetNextSqrtPriceFromAmount0RoundingUp() || denominator(%d) >= 0", denominator))
+	denominator := numerator1 - product
+
+	requirePositive(
```

Comment:
requirePositive checks whether `x > 0` while requireUnsigned checks for `x >= 0`
use requireUnsigned here